ffmpeg: fails to build

Maintainer: @thess, Espen Jürgensen
Environment: trunk, mips_mips32

Description:

ffmpeg fails to compile. See: https://downloads.openwrt.org/snapshots/faillogs/mips_mips32/packages/ffmpeg/mini/compile.txt

```
make -C /mnt/blue/openwrt/build_dir/target-mips_mips32_musl/ffmpeg-audio-dec/ffmpeg-3.2.10 DESTDIR="/mnt/blue/openwrt/build_dir/target-mips_mips32_musl/ffmpeg-audio-dec/ffmpeg-3.2.10/ipkg-install" all install
make[4]: Entering directory '/mnt/blue/openwrt/build_dir/target-mips_mips32_musl/ffmpeg-audio-dec/ffmpeg-3.2.10'
touch .version
mips-openwrt-linux-musl-gcc -I. -I./ -D_ISOC99_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200112 -D_XOPEN_SOURCE=600 -DPIC -DHAVE_AV_CONFIG_H -Os -pipe -mno-branch-likely -mips32 -mtune=mips32 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -iremap/mnt/blue/openwrt/build_dir/target-mips_mips32_musl/ffmpeg-audio-dec/ffmpeg-3.2.10:ffmpeg-3.2.10 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/mnt/blue/openwrt/staging_dir/target-mips_mips32_musl/usr/include -I/mnt/blue/openwrt/staging_dir/target-mips_mips32_musl/include -I/mnt/blue/openwrt/staging_dir/toolchain-mips_mips32_gcc-7.3.0_musl/usr/include -I/mnt/blue/openwrt/staging_dir/toolchain-mips_mips32_gcc-7.3.0_musl/include/fortify -I/mnt/blue/openwrt/staging_dir/toolchain-mips_mips32_gcc-7.3.0_musl/include -fpic   -std=c99 -fomit-frame-pointer -fPIC -mips32r6 -mips32r5 -mips32r2 -Wdeclaration-after-statement -Wall -Wdisabled-optimization -Wpointer-arith -Wredundant-decls -Wwrite-strings -Wtype-limits -Wundef -Wmissing-prototypes -Wno-pointer-to-int-cast -Wstrict-prototypes -Wempty-body -Wno-parentheses -Wno-switch -Wno-format-zero-length -Wno-pointer-sign -Wno-unused-const-variable -Os -fno-math-errno -fno-signed-zeros -fno-tree-vectorize -Werror=format-security -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=return-type -Werror=vla -Wformat -fdiagnostics-color=auto -Wno-maybe-uninitialized  -MMD -MF libavdevice/alldevices.d -MT libavdevice/alldevices.o -c -o libavdevice/alldevices.o libavdevice/alldevices.c
In file included from /mnt/blue/openwrt/staging_dir/toolchain-mips_mips32_gcc-7.3.0_musl/include/fortify/stdlib.h:20:0,
                 from ./libavutil/common.h:39,
                 from ./libavutil/avutil.h:290,
                 from ./libavutil/log.h:25,
                 from libavdevice/avdevice.h:48,
                 from libavdevice/alldevices.c:22:
./config.h:17:19: error: expected identifier or '(' before 'void'
 #define getenv(x) NULL
                   ^
./config.h:17:19: error: expected ')' before numeric constant
 #define getenv(x) NULL
                   ^
In file included from ./libavutil/internal.h:169:0,
                 from ./libavutil/common.h:467,
                 from ./libavutil/avutil.h:290,
                 from ./libavutil/log.h:25,
                 from libavdevice/avdevice.h:48,
                 from libavdevice/alldevices.c:22:
./libavutil/libm.h:54:32: error: static declaration of 'cbrt' follows non-static declaration
 static av_always_inline double cbrt(double x)
                                ^~~~
```

--- a/multimedia/ffmpeg/Makefile
+++ b/multimedia/ffmpeg/Makefile
@@ -9,12 +9,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=ffmpeg
-PKG_VERSION:=3.2.10
-PKG_RELEASE:=4
+PKG_VERSION:=3.2.12
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=https://ffmpeg.org/releases/
-PKG_HASH:=3c1626220c7b68ff6be7312559f77f3c65ff6809daf645d4470ac0189926bdbc
+PKG_HASH:=8985cea7b1b3b2e0e2b2a8ac6187a7fb022fe8aa9d35101760a000205c59c412
 PKG_MAINTAINER:=Ted Hess <thess@kitschensync.net>, \
 		Ian Leonard <antonlacon@gmail.com>
 
@@ -372,6 +372,9 @@ $(call Package/ffmpeg/Default/description)
 endef
 
 
+# Strip off FPU notation
+REAL_CPU_TYPE:=$(firstword $(subst +, ,$(CONFIG_CPU_TYPE)))
+
 FFMPEG_CONFIGURE:= \
 	CFLAGS="$(TARGET_CFLAGS) $(TARGET_CPPFLAGS) $(FPIC)" \
 	LDFLAGS="$(TARGET_LDFLAGS)" \
@@ -379,6 +382,7 @@ FFMPEG_CONFIGURE:= \
 	--enable-cross-compile \
 	--cross-prefix="$(TARGET_CROSS)" \
 	--arch="$(ARCH)" \
+	$(if $(REAL_CPU_TYPE),--cpu=$(call qstrip,$(REAL_CPU_TYPE)),) \
 	--target-os=linux \
 	--prefix="/usr" \
 	--pkg-config="pkg-config" \

